### PR TITLE
v2.3 Release Candidate

### PR DESCRIPTION
**Pull request by jeffhappily**
Note: Documentation fixes for https://docs.haskellstack.org/en/stable/ should target the "stable" branch, not master.

Please include the following checklist in your PR:

* [ ] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: package.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: stack
-version: '2.2.0'
+version: '2.3.0.0'
```
Comment (borsboom) [inline package.yaml]: For the RC, bump this to `2.3.0.1`.

File: ChangeLog.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
 
-## v2.3.x (unreleased changes)
+## v2.3.1 (unreleased changes)
```
Comment (borsboom) [inline ChangeLog.md]: For the RC, change this to `v1.9.0.1 (release candidate)`.

File: doc/appveyor.yml [hunk 3]
```diff
@@ -26,6 +26,8 @@ environment:
   - ARGS: "--resolver lts-11"
   - ARGS: "--resolver lts-12"
   - ARGS: "--resolver lts-13"
+  - ARGS: "--resolver lts-14"
```
[the remark this answers is not in the document]
Comment (jeffhappily) [reply doc/appveyor.yml]: It makes sense to me, I'll remove those

**Comment by jeffhappily** [comment]
> I believe we're dropping 32-bit Windows builds. If so, those should be removed from `doc/install_and_upgrade.md`

Do we have to remove it from the [releases.yaml](https://github.com/commercialhaskell/stackage-content/blob/master/stack/releases.yaml#L19-L22) as well?

**Comment by jeffhappily** [comment]
Here's the PR to bump the version to `v2.4` https://github.com/commercialhaskell/stack/pull/5241

**Comment by borsboom** [comment]
> > I believe we're dropping 32-bit Windows builds. If so, those should be removed from `doc/install_and_upgrade.md`
> 
> Do we have to remove it from the [releases.yaml](https://github.com/commercialhaskell/stackage-content/blob/master/stack/releases.yaml#L19-L22) as well?

No, it should be fine to leave there.  Requests to download it will just 404 because there's no binary available.

**Comment by borsboom** [comment]
Looks good.  Just awaiting @snoyberg's review of the example Travis configuration in the docs.

File: doc/travis-complex.yml [hunk 4]
```diff
@@ -66,6 +66,9 @@ jobs:
   - env: BUILD=cabal GHCVER=8.6.5 CABALVER=2.4 HAPPYVER=1.19.5 ALEXVER=3.1.7
     compiler: ": #GHC 8.6.5"
     addons: {apt: {packages: [cabal-install-2.4,ghc-8.6.5,happy-1.19.5,alex-3.1.7], sources: [hvr-ghc]}}
+  - env: BUILD=cabal GHCVER=8.8.3 CABALVER=3.0 HAPPYVER=1.19.12 ALEXVER=3.2.5
```
[the remark this answers is not in the document]
Comment (snoyberg) [reply doc/travis-complex.yml]: It's better to not include GHC 8.8.3 with cabal. The way building with cabal has changed significantly, and I never got it to work correctly. Personally, I've just stopped building with cabal on my CI. So I'd say: yes, include the updated resolvers for Stack, no to the cabal change.

[the remark this answers is not in the document]
Comment (jeffhappily) [reply doc/travis-complex.yml]: So you mean to keep GHC 8.8.3 and change Cabal back to 2.4?

[the remark this answers is not in the document]
Comment (snoyberg) [reply doc/travis-complex.yml]: No, remove this entry entirely. GHC 8.8.3 requires Cabal 3.0. And Cabal 3.0 requires too many changes to the rest of this file to be worth it.

**Comment by borsboom** [comment]
Merged this into master to avoid drift, but keeping the `rc/v2.3` branch open for further updates.

**Pull request merged**